src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -31,62 +18,55 @@ public Question(NsUser writer, String title, String contents) {
     }
 
     public Question(Long id, NsUser writer, String title, String contents) {
-        this.id = id;
-        this.writer = writer;
-        this.title = title;
-        this.contents = contents;
+        this(id, new DeleteRule(writer, "질문을 삭제할 권한이 없습니다."), new QuestionBody(title, contents));
     }
 
-    public Long getId() {
-        return id;
+    private Question(Long id, DeleteRule deleteRule, QuestionBody questionBody) {
+        super(id);
+        this.deleteRule = deleteRule;
+        this.questionBody = questionBody;
     }
 
     public String getTitle() {
-        return title;
+        return questionBody.getTitle();
     }
 
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
+    public void updateTitle(String title) {
+        this.questionBody=this.questionBody.updateTitle(title);
     }
 
     public String getContents() {
-        return contents;
-    }
-
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
+        return questionBody.getContents();
     }
 
-    public NsUser getWriter() {
-        return writer;
+    public void updateContents(String contents) {
+        this.questionBody=this.questionBody.updateContents(contents);
     }
 
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
         answers.add(answer);
     }
 
-    public boolean isOwner(NsUser loginUser) {
-        return writer.equals(loginUser);
+    public final ArrayList<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
```

Comment:
```suggestion
    public final void delete(NsUser loginUser) throws CannotDeleteException {
    public final List<DeleteHistory> toDeleteHistories() {
```

지금과 같이 삭제와 DeleteHistory 생성을 한 메서드에서 구현하는 것과 위와 같이 분리해서 구현하는 것의 장단점은 무엇일까?
어떻게 구현하는 것이 더 좋을까?